src/Services/Air/Air.js

```diff
@@ -45,6 +45,31 @@ module.exports = (settings) => {
       return service.gdsQueue(options);
     },
 
+    addSegments(options) {
+      if (
+        !options.version
+        || !options.universalRecordLocatorCode
+        || !options.reservationLocatorCode
+      ) {
+        return this.getBooking({ pnr: options.pnr })
+          .then((booking) => {
+            const missedOptions = {
+              version: options.version || booking.version,
+              reservationLocatorCode: (
+                options.reservationLocatorCode || booking.uapi_reservation_locator
+              ),
+              universalRecordLocatorCode: (
+                options.universalRecordLocatorCode || booking.uapi_ur_locator
+              ),
+            };
+
+            return service.addSegments(Object.assign(options, missedOptions));
```

Comment:
`Object.assign` is not necessary anymore, while you combine and object from single values above.
Also, `Object.assign` mutates first param.